src/pytorch_lightning/callbacks/progress/rich_progress.py

```diff
@@ -355,9 +381,11 @@ def on_train_epoch_start(self, trainer: "pl.Trainer", pl_module: "pl.LightningMo
     def on_validation_batch_start(
         self, trainer: "pl.Trainer", pl_module: "pl.LightningModule", batch: Any, batch_idx: int, dataloader_idx: int
     ) -> None:
-        if not self.has_dataloader_changed(dataloader_idx) or self.progress is None:
```

Comment:
Isn't this equal? When `is_disabled`, `self.progress` is None. Does that mean that there are times when this is disabled but progress exist? When does that happen normally?

Reply:
After the `on_x_start` hook runs, the value `is_disabled` and `self.progress is None` will always identical. The code change in this line is merely making it consistent with the other checks.

Reply:
The benefit of checking `self.progress is None` is that it avoids adding the `assertion` for mypy right after.

But if you prefer the current way, that's okay too. Feel free to resolve

Reply:
We don't actually "save" that many assertions. If you look at the usages in other hooks, we still need assertions like
`assert self.test_progress_bar_id is not None`. 
I prefer the expressiveness of `is_disabled`.